doc/config.md

```diff
@@ -605,13 +605,25 @@ setting is off (default), the `server_reset_query` will be run only in pools
 that are in sessions-pooling mode.  Connections in transaction-pooling mode
 should not have any need for a reset query.
 
-This setting is for working around broken setups that run applications that use session features
-over a transaction-pooled PgBouncer.  It changes non-deterministic breakage
+This setting is for working around ill-advised setups which run applications that use
+session features over a transaction-pooled PgBouncer.  It changes non-deterministic breakage
 to deterministic breakage: Clients always lose their state after each
-transaction.
+transaction. Client applications should be designed to not use session features
+over a transaction-pooled PgBouncer but this can provide a guardrail against
+unintentional breakage in an application or its' dependencies (for instance ActivRecord
+uses session features out of the box).
```

Comment:
I think it would be good to clarify exactly what the session level features are unsupported? Because a bunch of session features actually are supported these days, see `max_prepared_statements` and `track_extra_parameters`. So I think it would be good to call out what is and what isn't supported. And it might be useful to say what exactly ActiveRecord is doing that is unsupported.

Reply:
Do we really need to mention tools here? These tools might eventually change and this information is obsolete. I prefer the other way, open a PR for the referred product saying that it doesn't work with PgBouncer if you are using feature X. Once they fix this issue, their documentation is updated accordingly saying that PgBouncer is supported.